### PR TITLE
Convert plumbing.ErrObjectNotFound to git.ErrNotExist in getCommit

### LINKED ISSUE #10852: 500 Internal Server Error when accessing nonexistent commit
- Gitea version (or commit ref): 1.11.3
- Git version: 2.20.1
- Operating system: Debian Buster
- Database (use `[x]`):
  - [x] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [x] Yes: https://try.gitea.io/go-gitea/blog/commit/aa64449a8c91c7057c89322881794caf15590eb3
  - [ ] No
  - [ ] Not relevant
- Log gist: Not relevant

## Description

When accessing a commit that doesn't exist (either by typing nonsense into the url bar, or because it was later cleaned up by the garbage collector), the server fails with a 500 error, insead of a 404.

## Screenshots

https://try.gitea.io/img/500.png

### PR DESCRIPTION
**Pull request by zeripath**
Similar to a number of other bugs we need to handle plumbing.ErrObjectNotFound and convert to git.ErrNotExist appropriately.

Fix #10852

### CODE REVIEW

File: modules/git/repo_commit.go [hunk 1]
```diff
@@ -94,11 +94,18 @@ func (repo *Repository) getCommit(id SHA1) (*Commit, error) {
...
+				ID: id.String(),
+			}
+		}
 		if err == nil {
 			gogitCommit, err = repo.gogitRepo.CommitObject(tagObject.Target)
 		}
+		// if we get a plumbing.ErrObjectNotFound here then the repository is broken and it should be 500
 	}
 	if err != nil {
+
```
Comment (lunny) [inline modules/git/repo_commit.go]: Why we need a blank line here?

Comment (lafriks) [inline modules/git/repo_commit.go]:
```suggestion
```

**Review by lafriks — APPROVED** [review]

File: modules/git/repo_commit.go [hunk 2]
```diff
@@ -94,9 +94,15 @@ func (repo *Repository) getCommit(id SHA1) (*Commit, error) {
...
 		tagObject, err = repo.gogitRepo.TagObject(id)
+		if err == plumbing.ErrObjectNotFound {
+			return nil, ErrNotExist{
+				ID: id.String(),
+			}
+		}
 		if err == nil {
 			gogitCommit, err = repo.gogitRepo.CommitObject(tagObject.Target)
 		}
+		// if we get a plumbing.ErrObjectNotFound here then the repository is broken and it should be 500
```
Comment (guillep2k) [inline modules/git/repo_commit.go:105]: Although I'd prefer a "your repository is broken" error. 😉

**Review by guillep2k — APPROVED** [review]

**Comment by lafriks** [comment]
Please send backport

**Pull request merged**